index.d.ts

```diff
@@ -105,19 +105,24 @@ declare module 'spectacle' {
     size: number;
   }>;
 
+  type MdComponentProps = { [key: string]: any };
```

Comment:
I'm not thrilled about how loose this type is, but technically any props from valid MD components can be passed here, and the user can supply their own component map (in which case, we can't account for the prop types their components will accept). Seems like it'd be hard to make this as strict as possible - and making this unnecessarily strict is also a bad idea (since TS will yell about it not recognizing props that might actually be valid).

Reply:
I also think leaving it open gives us API flexibility. We could also use the `Record` type but I think that's two sides of the same coin here.